### ISSUE TITLE
[Bug] Toolbar scroll behavior causes improper page scroll

### ISSUE THREAD

**Issue by Cfun1** [xamarin/Xamarin.Forms#13338]
### Description
 I was trying to have a scroll behavior toolbar using a Shell custom renderer (code in repo).

It is working as expected, except that pages which before have their content fit the screen are now scrollable (by the amount of the toolbar height) while it is not needed nor desired of course, and the content at the end is down off screen.

### Steps to Reproduce
1. Run the repo.

### Expected Behavior
MainPage should not be scrollable, since it content fit the screen. In the contrary the scroll behavior of the toolbar should work on ListPage. 
### Actual Behavior
Notice that the content is not visible because down below screen edges, If you scroll from toolbar region than the page scrolls by the amount height of the toolbar.

### Basic Information

- Version with issue: 5.0.0.
- Last known good version: -
- Platform Target Frameworks: Tested on Android only
  - Android: 10.0 API 29

### Reproduction Link
Will be provided soon.

**Comment by Cfun1** [comment]
@hartez, I have included the repo.

**Comment by rmarinho** [comment]
hi @Cfun1  when i run the sample , i can't scroll on MainPage, do you mind include a video/screenshot for me to understand better.

Thanks

**Comment by Cfun1** [comment]
@rmarinho yes i will, in order to scroll you need to put the cursor/finger on the toolbar when you issue the swipe up gesture.

**Comment by rachelkang** [comment]
Hi, @Cfun1 - as @rmarinho mentioned, could you share a video if you're still experiencing the issue?
I tried to replicate the behavior described but also was unable to reproduce this issue. On my end, MainPage remains unscrollable whereas ListPage is scrollable

**Comment by Cfun1** [comment]
@rachelkang @rmarinho please check the screen record I have added.

**Comment by rachelkang** [comment]
@Cfun1 hmmm interesting.. thanks for sharing that! Which simulators/devices have you been able to reproduce this on?

**Comment by Cfun1** [comment]
@rachelkang I have only tested on Pixel 2 emulator so far.

**Comment by rachelkang** [comment]
Ohhh, I misunderstood the repro at first and didn't realize what you meant until now by scrolling up from the toolbar. I see it now! Very interesting, thanks for catching that @Cfun1 

@PureWeen this would be another good thing to keep in mind for Shell and multipage

**Comment by kcrg** [comment]
@Cfun1 Hi! I found a workaround for this issue. Just override native layout from Xamarin.Forms.Platform.Android ([BottomTabLayout.xml](https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Resources/Layout/BottomTabLayout.axml)) and add android:layout_marginBottom="56dp" (value from [official material design docs](https://material.io/components/bottom-navigation#specs)) to FrameLayout with id bottomtab.navarea.

Something like that:

<?xml version="1.0" encoding="utf-8"?>
    <androidx.coordinatorlayout.widget.CoordinatorLayout 
        xmlns:android="http://schemas.android.com/apk/res/android" 
        xmlns:app="http://schemas.android.com/apk/res-auto"
        android:layout_width="match_parent"
        android:layout_height="match_parent">

    <FrameLayout android:layout_marginBottom="56dp"
        android:id="@+id/bottomtab.navarea"
        android:layout_width="match_parent"
        android:layout_height="match_parent" 
        android:layout_gravity="fill"
        android:layout_weight="0" />

    <com.google.android.material.bottomnavigation.BottomNavigationView
        android:id="@+id/bottomtab.tabbar"
        android:theme="@style/Widget.MaterialComponents.BottomNavigationView"
        android:layout_gravity="bottom"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"/>
    </androidx.coordinatorlayout.widget.CoordinatorLayout>


I also found a way to add hiding Toolbar and BottomNavigationView on scroll behavior without writing custom Shell renderer but I think because it is (ToolBar and BottomNavigationView - [RootLayout.xml](https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Resources/Layout/RootLayout.axml) and [BottomTabLayout.xml](https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Resources/Layout/BottomTabLayout.axml)) in seperated CoordinatorLayouts, it works only with one at the same time. 
I spent the whole day figuring it out but in the end I think I must give up and stay with the Toolbar.
![1](https://user-images.githubusercontent.com/7996711/126363561-149283fa-519b-48a3-b9e3-f9d06b3a8259.PNG)


Additionaly if someone read this and want this behavior on BottomNavigationView or on top Toolbar without custom Shell renderer, here is how:

**BottomNavigationView:**
Override [BottomTabLayout.xml](https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Resources/Layout/BottomTabLayout.axml) in your Android project by simply adding this file to Layout folder in Resources and changing LinearLayout to CoordinatorLayout and putting app:layout_behavior="com.google.android.material.behavior.HideBottomViewOnScrollBehavior" property to BottomNavigationView control.

**Toolbar:**
Override  [RootLayout.xml](https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Resources/Layout/RootLayout.axml) just like in case above and put app:layout_scrollFlags="scroll|enterAlways" to Toolbar control.

Maybe something like that can be done also with classic navigation but I'm not 100% sure.

If anyone knows how to enable this to both, bottom Tabbar and top ActionBar, I'd love to know about it ;)
